tests/test_pyi_files.py

```diff
@@ -24,13 +24,18 @@ def test_pyi_file(path):
         if match:
             expected_output += f"{path}:{lineno}: {match.group(1)}\n"
 
-    # TODO: is a python version dependent error message really a good idea?
+    # TODO: are python-version-dependent error messages really a good idea?
     if sys.version_info < (3, 9):
         expected_output = re.sub(
             r'Y016 Duplicate union member ".*"',
             "Y016 Duplicate union member",
             expected_output,
         )
+        expected_output = re.sub(
+            r'Y019 Use "_typeshed\.Self" instead of ("_\w+"), e\.g\. "def .*?: \.\.\."',
+            r'Y019 Use "_typeshed\.Self" instead of \1',
```

Comment:
The problem was that you get an extra backslash, because the "replace with this" string doesn't need dots escaped:

```suggestion
            r'Y019 Use "_typeshed.Self" instead of \1',
```

Reply:
The line numbers also seem to be changed in the output, though. I can't work out what's causing that.